lib/src/util/yust_helpers.dart

```diff
@@ -271,4 +271,58 @@ class YustHelpers {
         return aString.compareTo(bString);
       });
   }
+
+  /// Formats a number to a string.
+  /// If [decimalCount] is null, the number is rounded to [maxPrecision] decimal places.
+  /// If [decimalCount] is given, the number is rounded to exactly [decimalCount] decimal places and optionally padded with zeros.
+  /// If [thousandsSeparator] is true, the number is formatted with a thousands separator.
+  /// Use [locale] to specify the thousands separator and decimal separator.
+  /// If [minWholeCount] is given, the number is padded with zeros to the left to reach at least [minWholeCount] digits.
+  /// The [minWholeCount] does not work with [thousandsSeparator] set to true.
+  String numToString(
+    num number, {
+    bool thousandsSeparator = false,
+    int minWholeCount = 1,
+    int? decimalCount,
+    int maxPrecision = 2,
+    String locale = 'de-DE',
+    String? unit,
+  }) {
+    assert((minWholeCount == 1) || !thousandsSeparator,
+        'minWholeCount does not work with thousandsSeparator set to true');
+    final wholePattern = thousandsSeparator ? '#,##0' : '0' * minWholeCount;
+    final decimalPattern =
+        decimalCount == null ? '#' * maxPrecision : '0' * decimalCount;
+    final format = NumberFormat(
+      '$wholePattern${decimalCount == 0 ? '' : '.'}$decimalPattern',
+      locale,
+    );
+    return unit == null || unit.isEmpty
+        ? format.format(number)
+        : '${format.format(number)} $unit';
+  }
+
+  /// Parse a string to a number.
+  /// If [maxPrecision] is null, the number is parsed as is.
+  /// If [maxPrecision] is 0, the number is rounded to the nearest integer.
+  /// If [maxPrecision] is given, the number is rounded to exactly [maxPrecision] decimal places.
+  /// Use [locale] to specify the thousands separator and decimal separator.
+  num? stringToNumber(String text,
```

Comment:
Try

String formatNumber(

  double value, {

  int leadingDigits = 6,

  int decimalPlaces = 4,

}) {

  // Build the pattern dynamically

  String leadingPart = List.filled(leadingDigits, '0').join(); // e.g. '000000'

  String decimalPart = decimalPlaces > 0 ? '.' + List.filled(decimalPlaces, '0').join() : '';

  String pattern = '$leadingPart$decimalPart';
 
  // Add grouping separators

  final formatter = NumberFormat.decimalPattern()

    ..minimumIntegerDigits = leadingDigits

    ..minimumFractionDigits = decimalPlaces

    ..maximumFractionDigits = decimalPlaces;
 
  return formatter.format(value);

}